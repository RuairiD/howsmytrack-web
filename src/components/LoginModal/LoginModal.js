import React from 'react';

import { Modal } from 'antd';
import LoginForm from '../LoginForm/LoginForm';

type Props = {
    onCancel: () => void,
    isVisible: boolean,
};

class LoginModal extends React.Component<Props> {
    /*
     * Component for displaying modal containing LoginForm for users to sign in.
     */

    render() {
        return (
            <Modal
                title="Sign In"
                onCancel={this.props.onCancel}
                visible={this.props.isVisible}
                footer={null}
                destroyOnClose
                className="responsive-modal"
            >
                <LoginForm />
            </Modal>
        );
    }
}

export default LoginModal;
