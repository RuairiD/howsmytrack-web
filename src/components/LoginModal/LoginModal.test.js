import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import LoginModal from "./LoginModal";

describe("LoginModal", () => {
    it("renders a Modal containing a LoginForm", () => {
        const onCancel = () => {};
        const wrapper = shallow(
            <LoginModal
                onCancel={onCancel}
                isVisible
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
