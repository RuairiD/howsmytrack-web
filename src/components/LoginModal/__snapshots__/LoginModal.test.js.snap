// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginModal renders a Modal containing a LoginForm 1`] = `
<Modal
  className="responsive-modal"
  confirmLoading={false}
  destroyOnClose={true}
  footer={null}
  maskTransitionName="fade"
  okType="primary"
  onCancel={[Function]}
  title="Sign In"
  transitionName="zoom"
  visible={true}
  width={520}
>
  <Form(UnwrappedLoginFormContainer) />
</Modal>
`;
